lib/local_daily_digest.js

```diff
@@ -0,0 +1,167 @@
+var async = require('async'),
+	_ = require('lodash'),
+	log = require('../log'),
+	Profile = require('../models').Profile,
+	Contact = require('../models').Contact,
+	log = require('../log'),
+	mail = require('../mail');
+
+function get(req, res, next) {
+	var result = {},
+	 	allLocalProfilesWDigest = [],
+	 	allLocalProfiles =[],
+	 	requiredUsers=[],
+	 	checkedOutUsers = [],
+	 	finalCheckedOutList=[],
+		tempManagers =[],
+		tempUsers = [],
+		managerContactsProfile = [],
+		finalOutput = [],
+		tempProf=[],
+		requiredLocationsId =[],
+		requiredLocationsName =[],
+		temp='',
+		tempProfile ='',
+		checkedInUsers = [];
+	
+	async.series([
+		//THe local daily digest will be just the digest for that country !!
+		function (cb) {
+			Profile.find({'dailyDigest': {$not: {$size: 0}}}, function(err, profile){
+			if(err){
+		  		res.send(500, new Error(err));
+		  		return cb(true);
+		  	}
+
+		  	if(!profile){
+		  		res.send(400, new Error('No Profiles were found'));
+		  	}
+
+		  	profile.forEach(function(profile){
+			  		allLocalProfilesWDigest.push(profile);
+			  	});
+	  			return cb();
+			});
+		},
+		function(cb){
+			allLocalProfilesWDigest.forEach(function(profile){
+				// console.log(profile);
+				profile.dailyDigest.forEach(function(location){
+					requiredLocationsId.push(location);
+				});
+				// requiredLocationsName.push(profile.location);
+			});
+			console.log("the countries this profile is either a manger or editor of are: ", requiredLocationsId);
+			return cb();
+		},
+		function (cb){
+			Contact.find({'type':'local'}, function(err, contact){
+				if (err) {
+		          res.send(500, new Error("err"));
+		          return cb(true);
+		        }
+		        if (!contact) {
+		          res.send(404, new Error('No Local Profiles found  not found'));
+		          return cb(true);
+		        }
+			contact.forEach(function(profile){
+				allLocalProfiles.push(profile);
+			});
+	  		return cb();
+		  });
+		},
+		function (cb){
+			allLocalProfilesWDigest.forEach(function(managerProfile){
+				allLocalProfiles.forEach(function(profile){
+					if(managerProfile.locationId == profile.locationId){
```

Comment:
Not sure how this is going to work since allLocalProfilesWDigest contains Profile items (which don't have any locationId)